server/src/test/java/org/candlepin/service/impl/stub/StubEntitlementCertServiceAdapter.java

```diff
@@ -87,4 +91,31 @@ public EntitlementCertificate generateUeberCert(Entitlement entitlement,
         return generateEntitlementCert(entitlement, product);
     }
 
+    @Override
+    public Map<String, EntitlementCertificate> generateEntitlementCerts(Consumer consumer,
+            Map<String, Entitlement> entitlements, Map<String, Product> products)
+        throws GeneralSecurityException, IOException {
+
+        Map<String, EntitlementCertificate> result = new HashMap<String, EntitlementCertificate>();
+        for (Entry<String, Entitlement> entry : entitlements.entrySet()) {
+            EntitlementCertificate cert = generateEntitlementCert(entry.getValue(),
+                    products.get(entry.getKey()));
+            result.put(entry.getKey(), cert);
+        }
+        return result;
+    }
+
+    @Override
+    public Map<String, EntitlementCertificate> generateUeberCerts(Consumer consumer,
+            Map<String, Entitlement> entitlements, Map<String, Product> products)
+        throws GeneralSecurityException, IOException {
```

Comment:
@awood 
if(you mean to suggest that we change the interface too) {
* I disagree.
 *  Even if we may not catch this specific exception for this method in all the implementations of this interface, I think its safer to have specific exceptions at the cost of additional imports. Consider a developer making changes such that another specific kind of exception is thrown. The developer may miss that, if we throw the broader exception here. there may be value in catching that specific exception, if not any of the current ones.
 *  Besides, there are some example of code where we do catch this specific exception also ( ProductCertificatCurator )

}  else    {
    cant be done. can not throw broader exceptions in overriding methods
}